pkg/asset/agent/installconfig_test.go

```diff
@@ -392,6 +392,45 @@ pullSecret: "{\"auths\":{\"example.com\":{\"auth\":\"authorization value\"}}}"
 			expectedFound: false,
 			expectedError: "invalid install-config configuration: Platform: Invalid value: \"baremetal\": CPU architecture \"ppc64le\" only supports platform \"none\".",
 		},
+		{
+			name: "invalid platform.baremetal for architecture s390x",
+			data: `
+apiVersion: v1
+metadata:
+  name: test-cluster
+baseDomain: test-domain
+networking:
+  networkType: OVNKubernetes
+  machineNetwork:
+  - cidr: 192.168.122.0/23
+compute:
+  - architecture: s390x
+    hyperthreading: Enabled
+    name: worker
+    platform: {}
+    replicas: 0
+controlPlane:
+  architecture: s390x
+  hyperthreading: Enabled
+  name: master
+  platform: {}
+  replicas: 3
+platform:
+  baremetal:
+    apiVIP: 192.168.122.10
+    ingressVIP: 192.168.122.11
+    hosts:
+    - name: host1
+      bootMACAddress: 52:54:01:aa:aa:a1
+    - name: host2
+      bootMACAddress: 52:54:01:bb:bb:b1
+    - name: host3
+      bootMACAddress: 52:54:01:cc:cc:c1
+pullSecret: "{\"auths\":{\"example.com\":{\"auth\":\"authorization value\"}}}"
+`,
+			expectedFound: false,
+			expectedError: "invalid install-config configuration: Platform: CPU architecture \"s390x\" only supports platform \"baremetal\".",
```

Comment:
It's not clear to me what this is testing. Is baremetal platform a valid one for s390x? The test case name says baremetal is not a valid platform for s390x, but expectedError says the opposite.

Reply:
> Is baremetal platform a valid one for s390x?

It is not as far as I know.

Reply:
changed it to platform type none